File: src/algorithms/stacks-queues/stack-applications/remove-all-adjacent-duplicates/__tests__/RemoveAllAdjacentDuplicates_test.java
// javac RemoveAllAdjacentDuplicates.java RemoveAllAdjacentDuplicates_test.java && java -ea RemoveAllAdjacentDuplicates_test
public class RemoveAllAdjacentDuplicates_test {
    public static void main(String[] args) {
        assert RemoveAllAdjacentDuplicates.removeAllAdjacentDuplicates("abbaca").equals("ca");
        assert RemoveAllAdjacentDuplicates.removeAllAdjacentDuplicates("azxxzy").equals("ay");
        assert RemoveAllAdjacentDuplicates.removeAllAdjacentDuplicates("").equals("");
        assert RemoveAllAdjacentDuplicates.removeAllAdjacentDuplicates("abc").equals("abc");
        assert RemoveAllAdjacentDuplicates.removeAllAdjacentDuplicates("aaaaaa").equals("");
        assert RemoveAllAdjacentDuplicates.removeAllAdjacentDuplicates("aabb").equals("");
        assert RemoveAllAdjacentDuplicates.removeAllAdjacentDuplicates("a").equals("a");
        assert RemoveAllAdjacentDuplicates.removeAllAdjacentDuplicates("abba").equals("");

        System.out.println("All tests passed!");
    }
}

File: src/algorithms/stacks-queues/stack-applications/remove-all-adjacent-duplicates/__tests__/RemoveAllAdjacentDuplicates_test.cpp
// g++ -o RemoveAllAdjacentDuplicates_test RemoveAllAdjacentDuplicates_test.cpp && ./RemoveAllAdjacentDuplicates_test
#define TESTING
#include "../sources/RemoveAllAdjacentDuplicates.cpp"
#include <cassert>
#include <iostream>
#include <string>

int main() {
    assert(removeAllAdjacentDuplicates("abbaca") == "ca");
    assert(removeAllAdjacentDuplicates("azxxzy") == "ay");
    assert(removeAllAdjacentDuplicates("") == "");
    assert(removeAllAdjacentDuplicates("abc") == "abc");
    assert(removeAllAdjacentDuplicates("aaaaaa") == "");
    assert(removeAllAdjacentDuplicates("aabb") == "");
    assert(removeAllAdjacentDuplicates("a") == "a");
    assert(removeAllAdjacentDuplicates("abba") == "");

    std::cout << "All tests passed!" << std::endl;
    return 0;
}
